client/signup/steps/plans/index.jsx

```diff
@@ -81,13 +81,21 @@ module.exports = React.createClass( {
 		analytics.tracks.recordEvent( 'calypso_signup_compare_plans_click', { location: linkLocation } );
 	},
 
+	hideFreePlan: function() {
+		if ( this.props.signupDependencies && this.props.signupDependencies.domainItem ) {
+			return this.props.signupDependencies.domainItem.is_domain_registration;
+		}
+
+		return this.props.hideFreePlan;
+	},
```

Comment:
Overwriting props depending on something else makes things quite complicated very fast. Is it possible to move this to a prop coming from the parent component?